app.py

```diff
@@ -43,6 +44,12 @@ def create_app() -> FlaskApp:
     health = Healthcheck(flask_app)
     health.add_check(FlaskRunningChecker())
     health.add_check(DbChecker(db))
+
+    @flask_app.errorhandler(404)
+    def not_found(error):
+        flask_app.logger.warning(f"requested URL was not found on the server")
+        return f"{str(error)}", 404
```

Comment:
I don't think it's valuable to have this log message. We should already get a response log from gunicorn/uvicorn that has the status code and the URL, so this just feels extraneous.

Also, I don't think we should return the error message directly. We should probably return JSON in some "semi-standard" format: `{"code": 404, "error": "Not found"}` feels appropriate?

Reply:
I think better to have a generalise  way to do this so cant we have this error handling on utils then we can use it for other services as well and no need to worry about the error message structure ?

Reply:
yes, agree. Wouldn't hurt for fsd-utils to have some standard error handlers potentially (surprised that connexion doesn't offer some out of the box?)

Reply:
@samuelhwilliams / @nuwan-samarasinghe : currently there are error handlers for the frontend /Authenticator in the apps itself rendering the html pages 404/500.html. May be we can create ticket for adding error handlers for the backend apps. like fund-store/Application and assessment-store . Also we can define a uniform wrapper to render the error message in a specific format